DotNetDBF/DBFReader.cs

```diff
@@ -245,175 +244,156 @@ internal Object[] NextRecord(IEnumerable<int> selectIndexes, IList<int> sortedIn
+                                        int.TryParse(monthString, out month) &&
+                                        int.TryParse(dayString, out day))
+                                    {
+                                        recordObjects[i] = new DateTime(year, month, day);
+                                    }
+                                    else
+                                    {
+                                        recordObjects[i] = null;
+                                    }
                                 }
-                                else
+                                catch (ArgumentOutOfRangeException)
                                 {
+                                    /* this field may be empty or may have improper value set */
                                     recordObjects[i] = null;
                                 }
-                            }
-                            catch (ArgumentOutOfRangeException)
-                            {
-                                /* this field may be empty or may have improper value set */
-                                recordObjects[i] = null;
-                            }
-
-                            break;
 
+                                break;
+                            }
                         case NativeDbType.Float:
-
-                            try
                             {
-                                byte[] t_float = new byte[
-                                    _header.FieldArray[i].FieldLength
-                                    ];
-                                _dataInputStream.Read(t_float, 0, t_float.Length);
-                                String tParsed = CharEncoding.GetString(t_float);
-                                var tLast = tParsed.Substring(tParsed.Length - 1);
-                                if (tParsed.Length > 0
-                                    && tLast != " "
-                                    && tLast != DBFFieldType.Unknown)
+                                try
                                 {
-                                    recordObjects[i] = Double.Parse(tParsed, NumberStyles.Float | NumberStyles.AllowLeadingWhite);
+                                    byte[] t_float = new byte[_header.FieldArray[i].FieldLength];
+                                    _dataInputStream.Read(t_float, 0, t_float.Length);
+                                    string tParsed = CharEncoding.GetString(t_float);
+                                    string tLast = tParsed.Substring(tParsed.Length - 1);
+                                    if (tParsed.Length > 0
+                                        && tLast != " "
+                                        && tLast != DBFFieldType.Unknown)
+                                    {
+                                        recordObjects[i] = Convert.ToSingle(tParsed, NumberFormatProvider);

```

Comment:
This is the part I have changed. You see, I had a case when value in DBF table was "0.436". But because of default culture settings `float.Parse` expected decimal separator to be a comma (because it's a default behavior in my culture). Then it throw an exception.